### PR TITLE
[JENKINS-73744] npm scripts lint:fix do not work 

### PR DESCRIPTION
**Pull request by scherler**
See [JENKINS-73744](https://issues.jenkins.io/browse/JENKINS-73744).

### Proposed changelog entries

- human-readable text

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by timja — COMMENTED** [review]
Please update the `CONTRIBUTING` guide: https://github.com/jenkinsci/jenkins/blob/master/CONTRIBUTING.md#running-the-yarn-frontend-build

Changes look good although I would like to run locally before approving

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -110,17 +102,15 @@ To automatically fix backend issues, run:
 mvn spotless:apply
 ```
 
-To view frontend issues, after [adding Node and Yarn to your path](#running-the-yarn-frontend-build), run:
+To view frontend issues run:
```
Comment (janfaracik) [inline CONTRIBUTING.md:105]:
```suggestion
To view frontend issues, run:
```

Missing a comma

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -110,17 +102,15 @@ To automatically fix backend issues, run:
...
 
 ```sh
-cd war
 yarn lint
 ```
 
-To fix frontend issues, after [adding Node and Yarn to your path](#running-the-yarn-frontend-build), run:
+To fix frontend issues run:
```
Comment (janfaracik) [inline CONTRIBUTING.md:111]:
```suggestion
To fix frontend issues, run:
```

Missing a comma

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -75,10 +68,9 @@ On one terminal, start a development server that will not process frontend asset
 MAVEN_OPTS='--add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.io=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED' mvn -pl war jetty:run -Dskip.yarn
 ```
 
-On another terminal, move to the `war` folder and start a [webpack](https://webpack.js.org/) dev server, after [adding Node and Yarn to your path](#running-the-yarn-frontend-build):
+On another terminal and start a [webpack](https://webpack.js.org/) dev server:
```
Comment (janfaracik) [inline CONTRIBUTING.md:71]:
```suggestion
Open another terminal and start a [webpack](https://webpack.js.org/) dev server:
```

**Comment by janfaracik** [comment]
Working locally for me +1 - ran a dev server and tried linting.

I think it makes more sense for first time contributors to have `package.json` in the root too.

**Comment by scherler** [comment]
https://github.com/scherler/jenkins/pull/75 I created a PR for the profiles only but I am not sure if that is what @timja had in mind

**Review by timja — APPROVED** [review]
Thanks! I tested locally and linting looks to be fine for me.

**Comment by MarkEWaite** [comment]
I found that with the most recent changes and a merge of the latest master branch, I needed to add the downloaded versions of Node and Yarn to my path.  I did that with the command:

```
export PATH=$PWD/node:$PWD/node/yarn/dist/bin:$PATH
```

I have no skills in front-end development, so that may indicate that I'm making some obvious mistake, but when I tried to run `yarn` without adding it to the PATH, my Linux shell reported command not found.

Once the PATH was updated, I confirmed that `yarn lint:fix` worked for me when I intentionally modified a JavaScript file to have a formatting error.

**Comment by timja** [comment]
That’s expected, maven doesn’t add the downloaded yarn and node to the PATH.

**Comment by MarkEWaite** [comment]
> That’s expected, maven doesn’t add the downloaded yarn and node to the PATH.

Thanks!  I think that means the CONTRIBUTING doc should be updated to include the instructions to set the PATH for the new location of node and yarn.  Any objections if I push that change to the CONTRIBUTING file?

**Review by MarkEWaite — APPROVED** [review]

**Comment by daniel-beck** [comment]
In what way does this address JENKINS-73744?

**Comment by timja** [comment]
@scherler are you happy with that change?

I and most other people doing frontend development are unlikely to use that approach as we'll use what we have locally installed so we don't have to modify our PATH for one project.

Node will read what's in the package.json and automatically load the right version of yarn anyway once you've run `corepack enable`

**Comment by MarkEWaite** [comment]
> I and most other people doing frontend development are unlikely to use that approach as we'll use what we have locally installed so we don't have to modify our PATH for one project.

Would it be a workable compromise for the less experienced (like me) if there were a section that described the PATH setting without including a link in the later sections to those instructions?

**Comment by scherler** [comment]
@MarkEWaite @timja I will update doc to point out the `path` facts

**Comment by scherler** [comment]
@daniel-beck I am not sure about the question since the change is addressing the problem and it is now possible to use `lint:fix` as expected and without having to add the node_modules to the path

**Comment by basil** [comment]
> @daniel-beck I am not sure about the question since the change is addressing the problem and it is now possible to use `lint:fix` as expected and without having to add the node_modules to the path.

Then why does `CONTRIBUTING.md` still talk about adding to the PATH?

**Comment by basil** [comment]
And why was the `node.version` configuration in `.github/renovate.json` not updated to adjust to the new file path?

**Comment by scherler** [comment]
@basil Actually @MarkEWaite added that note, you only need to add node/yarn to your path if you want to use `yarn ...` in your jenkins setup without having `yarn` globally installed in your box. In my case I do not have to do this anymore since I have yarn installed on a global level. Hence https://github.com/jenkinsci/jenkins/pull/9718#issuecomment-2352308547 `you would have yarn and node in your path anyway but I guess you than just ignore the note.`  I was not aware of that config `.github/renovate.json` but we neither changed  the version, but I think you mean

**Comment by basil** [comment]
@scherler The documentation should always reflect what is supported by the project, no more and no less.

- If usage with `PATH` modification is supported and usage without `PATH` modification is unsupported, then this PR should be reverted.
- If usage with `PATH` modification is unsupported and usage without `PATH` modification is supported, then the documentation should be updated to remove references to `PATH` modification.
- If usage with or without `PATH` modification is supported, the documentation should describe both usages.

In any case, this PR (as it was integrated) is logically inconsistent.

**Comment by scherler** [comment]
@basil https://github.com/jenkinsci/jenkins/pull/9757

**Comment by basil** [comment]
> so not really sharing your opinion of `logically inconsistent`

Is there any particular reason to use monospaced text here for the phrase "logically inconsistent"? Did you mean to use [quotation marks](https://www.grammarly.com/blog/quotation-marks/)?

File: pom.xml [hunk 4]
```diff
@@ -415,5 +421,192 @@ THE SOFTWARE.
...
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+    <profile>
+      <id>yarn-lint</id>
+      <activation>
+        <property>
+          <name>!env.CI</name>
```
[the remark this answers is not in the document]
Comment (MarkEWaite) [reply pom.xml:460]: I should have added notes in the PR description on the testing that I did in the [comment](https://github.com/jenkinsci/jenkins/pull/9718#issuecomment-2350690049).  Tim noted in his comment that he [tested locally](https://github.com/jenkinsci/jenkins/pull/9718#pullrequestreview-2300229313), and Jan noted in his comment that he [tested locally](https://github.com/jenkinsci/jenkins/pull/9718#issuecomment-2345660780).  

However, I did not test the Maven profile and should have done that.

**Pull request merged**